django/university/routes/MarketplaceExchangeView.py

```diff
@@ -0,0 +1,186 @@
+                    queryset=MarketplaceExchangeClass.objects.all(),
+                    to_attr='options'
+                )
+            ).filter(issuer_nmec=request.user.username).all())
+
+        if course_unit_name_filter:
+            marketplace_exchanges = list(filter(
+                lambda x: self.courseUnitNameFilterInExchangeOptions(x.options, course_unit_name_filter),
+                marketplace_exchanges
+            ))
+
+        return self.build_pagination_payload(request, marketplace_exchanges)
+
+    def filterAllExchanges(self, request, course_unit_name_filter, classes_filter):
+        print("classes filter: ", classes_filter)
+        # courseUnitClasses = StudentScheduleView.retrieveCourseUnitClasses(SigarraController(), request.user.username)
+        marketplace_exchanges = list(MarketplaceExchange.objects
+                .exclude(issuer_nmec=request.user.username).prefetch_related(
+                Prefetch(
+                    'marketplaceexchangeclass_set',
+                    # queryset=MarketplaceExchangeClass.objects.filter(
+                    #     reduce(lambda x, y: x | y, [
+                    #         Q(class_issuer_goes_to=k) 
+                    #         for k, v in courseUnitClasses.items()
+                    #     ])
+                    # ),
+                    to_attr='options'
+                )
+            ).all())
+
+        marketplace_exchanges = self.advanced_classes_filter(marketplace_exchanges, classes_filter)
+        
+        if course_unit_name_filter:
+            marketplace_exchanges = list(filter(
+                lambda x: self.courseUnitNameFilterInExchangeOptions(x.options, course_unit_name_filter),
+                marketplace_exchanges
+            ))
+
+        return self.build_pagination_payload(request, marketplace_exchanges)
+
+    def advanced_classes_filter(self, marketplace_exchanges, classes_filter):
+        filtered_marketplace_exchanges = []
+        for exchange in marketplace_exchanges:
+            exchange_tainted = False
+            for option in exchange.options:
+                if option.course_unit_acronym in classes_filter.keys():
+                    correct_class_included = option.class_issuer_goes_from in classes_filter[option.course_unit_acronym]
+                    exchange_tainted = not correct_class_included
+
+            if not exchange_tainted:
+                filtered_marketplace_exchanges.append(exchange)
+
+        return filtered_marketplace_exchanges
+
+    def courseUnitNameFilterInExchangeOptions(self, options, courseUnitNameFilter):
+        matches = []
+        for courseUnitId in courseUnitNameFilter:
+            for option in options:
+                if courseUnitId == option.course_unit_id:
+                    matches.append(1)
+
+        return len(matches) == len(courseUnitNameFilter)
+   
+
+    """
+        Returns all the current marketplace exchange requests paginated
+    """
+    def get(self, request):
```

Comment:
Can this method name be more specific?

Using documentation may indicate the method name should be improved

Reply:
The name has to be `get` as it is the function responsible for handling `GET` requests to the endpoint this view is responsible for.

Maybe it is better to remove the documentation comment instead?

Reply:
I understand. The documentation is right then.